app/controllers/users_controller.rb

```diff
@@ -1,4 +1,5 @@
-class UsersController<ApplicationController
+class UsersController< ApplicationController
+  before_action :require_user, except: [:new, :create]
 
```

Comment:
Does this level of authorization still let visitors browse the site (items and merchants) and add items to the cart?

Reply:
Yes, this only limits visitor access to the user#show/#edit/#destroy (user profile features). Visitor can still add items to cart and view merchants/items